packages/react/src/components/accordion/accordion-container.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, ReactElement, createRef, useRef } from 'react';
+import { AccordionProps, AccordionContainerProps } from './accordion-types';
+import { Accordion } from './accordion';
+import { StyledAccordionGroup } from './accordion-styles';
+
+export const AccordionContainer: React.FC<AccordionContainerProps> = ({
+    id, mode = 'single', children,
+}) => {
+    const generateId = (childProps: AccordionProps, index: number): string => childProps.id || `${id}-${index}`;
+    const isAccordion = (child: React.ReactNode): child is ReactElement<AccordionProps> => (
+        React.isValidElement<AccordionProps>(child) && child.type === Accordion);
+    const focusedAccordionRef = useRef<number | null>(null);
+
+    const [expandedItemIds, setExpandedItemIds] = useState<string[]>(() => (
+        React.Children.toArray(children)
+            .filter(isAccordion)
+            .map((child: ReactElement<AccordionProps>, index) => {
+                const childProps = child.props as AccordionProps;
+                return childProps.isExpanded ? generateId(childProps, index) : null;
+            })
+            .filter((expandedId) => expandedId !== null) as string[]
+    ));
+
+    const handleToggle = (itemId: string): void => {
+        if (mode === 'single') {
+            setExpandedItemIds((prevIds) => (prevIds.includes(itemId) ? [] : [itemId]));
+        } else if (mode === 'multi') {
+            setExpandedItemIds((prevIds) => {
+                if (prevIds.includes(itemId)) {
+                    return prevIds.filter((expandedId) => expandedId !== itemId);
+                }
+                return [...prevIds, itemId];
+            });
+        }
+    };
+
+    const childrenArray: ReactElement<AccordionProps>[] = React.Children.toArray(children)
+        .filter(isAccordion)
+        .map((child: ReactElement<AccordionProps>, index) => {
+            const buttonRef = createRef<HTMLButtonElement>();
```

Comment:
Some concerns with this ref.

- In function component, we should use `useRef` instead of createRef. 
- The ref should point to the same variable between refresh, while this code create a new ref every time.
- This ref itself has no use. You could forward a ref from the user to the target element, but if you create a ref yourself, it should have some code that use that ref (like some read/write on it).

Reply:
Maybe just remove if not needed

Reply:
@savutsang I did a adjustment in the code to avoid regenerate ref between refresh. Let me explain some decisions I took to make sense on my refactor:
1. I'm using useMemo to ensures that the button refs are only created once and persist across renders, avoiding unnecessary re-creation.
2. With this implementation I can't use useRef as it's not allowed to use built-in hooks inside useMemo.

Note:
I saw the same implementation in tabs which is using useMemo with createRef in a functional component.
using createRef() within useMemo() seems to be valid and will create the refs once.

Reply:
New code and clarification looks good to me.